FirstHomeWork

RinHasNoArms opened the pull request.

RinHasNoArms committed:
hw

YuriyBiznigaev commented:
@RinHasNoArms 
1. установить min версию  iOS Deployment Target = 14.0 в настройках проекта
![image](https://user-images.githubusercontent.com/75571475/159989006-83b00bfd-8102-49f4-b5af-42025abb486d.png)
2. добавить отступ между карточками потому что сейчас не понятно где заканчивается предыдущая и начинается следующая 
<img width="442" alt="Screenshot 2022-03-24 at 21 57 15" src="https://user-images.githubusercontent.com/75571475/159990783-29673b5c-8edf-45f1-b7c6-71d51d16bc9c.png">
3. доработать механику двойного нажатия на изображение для анимированного показа сердца поверх картинки:
сердце не должно показывается если происходит уменьшение значения лайков / отмена лайка

RinHasNoArms commented:
По п3 я же могу показывать анимацию пустого сердца или лучше вообще убрать?

RinHasNoArms commented:
> @RinHasNoArms
> 
> 2. добавить отступ между карточками потому что сейчас не понятно где заканчивается предыдущая и начинается следующая
> 
> <img alt="Screenshot 2022-03-24 at 21 57 15" width="442" src="https://user-images.githubusercontent.com/75571475/159990783-29673b5c-8edf-45f1-b7c6-71d51d16bc9c.png">
> 
> 3. доработать механику двойного нажатия на изображение для анимированного показа сердца поверх картинки: сердце не должно показывается если происходит уменьшение значения лайков / отмена лайка

По п2, как лучше сделать? Я могу добавить вьюху в нижнюю часть ячейки и поставить зависимость от ее высоты или же увеличить отступ от нижней границы ячейки у лейбла с текстом

По п3 Можно поставить анимацию с не закрашенным сердечком или же совсем убрать?

stayn commented on hw1/homeWork/Extensions/Extensions.swift:
```diff
@@ -0,0 +1,44 @@
+            likes = "лайка"
+        case 5...9:
+            likes = "лайков"
+        default:
```
Про склонение числительных [тут](https://medium.com/@vitaliikuznetsov/%D0%BB%D0%BE%D0%BA%D0%B0%D0%BB%D0%B8%D0%B7%D0%B0%D1%86%D0%B8%D1%8F-%D0%BC%D0%BD%D0%BE%D0%B6%D0%B5%D1%81%D1%82%D0%B2-%D0%B8%D0%BC%D0%B5%D0%BD-%D1%81%D1%83%D1%89%D0%B5%D1%81%D1%82%D0%B2%D0%B8%D1%82%D0%B5%D0%BB%D1%8C%D0%BD%D1%8B%D1%85-%D0%B2-ios-463dfe69048b)

stayn commented on hw1/homeWork/MVP/Model/Constants.swift:
```diff
@@ -0,0 +1,34 @@
+    static let seventh = UIImage(named: "seventhImg")!
+    static let eighth = UIImage(named: "eighthImg")!
+    static let ninth = UIImage(named: "ninthImg")!
+    static let tenth = UIImage(named: "tenthImg")!
```
Force unwrapping — штука опасная. Лучше оставлять optional, а потом разворачивать через `if let` или `guard`. Ну и должны ли быть все эти картинки static?

stayn commented on hw1/homeWork/ViewController.swift:
```diff
@@ -0,0 +1,19 @@
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view.
+    }
```
А зачем нам тут пустой VC? :)

RinHasNoArms replied:
В самом проекте его нет, из папки не удалил

stayn commented on hw1/homeWork/mainViewController.swift:
```diff
@@ -0,0 +1,82 @@
+
+import UIKit
+
+final class mainViewController: UIViewController, UITableViewDelegate, UITableViewDataSource, CustomCellDelegate, PostPresenterDelegate {
```
Обычно для названия классов используется UpperCamelCase, можно поподробнее изучить какой-нибудь [style guide](https://github.com/raywenderlich/swift-style-guide).

RinHasNoArms committed:
Исправления

1. В проекте установлена min версия ios =14
2. Добавил сепаратор чтобы был отступ между карточками
3. Доработал механику двойного нажатия, теперь нет анимации сердечка при уменьшении значения лайков
4. Исправлены склонения числительных
5. Убрана вьюшка из папки проекта
6. Нейминг классов

stayn commented on hw1/homeWork/MVP/Presenter/PostPresenter.swift:
```diff
@@ -29,17 +29,27 @@ class PostPresenter {
-        let post9 = Post(imageProfile: imagesProfiles.ninth, nameProfile: "Dmitry Grunenkov", imagePost: imagesPosts.ninth, textProfile: "#35mm #vsco #vscocam #vscorussia #nikon #nikonrussia #инстаграмнедели #natgeoru #natgeo #msk #mskpit #autumn #leaf #red #sun #sunlight", countLikes: UInt.random(in: 0..<1000))
-        let post10 = Post(imageProfile: imagesProfiles.tenth, nameProfile: "Isaac Lane Koval", imagePost: imagesPosts.tenth, textProfile: "#vsco #vscocam #vscorussia #nikon #bigstonebridge #cathedralofchristthesaviour #night #moscow", countLikes: UInt.random(in: 0..<1000))
+    func fetchData() -> [Post]? {
+        guard let imageProfileFirst = ImagesProfiles.image.first else {return nil}
```
Если одного из изображений не будет, то функция вернет nil вместо всех данных, это точно так было запланировано? Ну и сначала `guard let imageProfileThird = ImagesProfiles.image.third`, а потом всё равно используется `ImagesPosts.image.third!`))) Можно сделать так, чтоб в `Post` можно было передавать `imagePost` в виде `optional`, а дальше уже по желанию — в `UIImageView.image` можно передавать `UIImage?` (ну или при установке изображения в UIImageView проверять что изображение не nil и в обратном случае подставлять заглушку, например)

RinHasNoArms replied:
С этим еще не разобрался до конца как лучше сделать

RinHasNoArms replied:
спасибо за совет

RinHasNoArms committed:
Исправления

imageProfile optional в Post

RinHasNoArms committed:
comment

RinHasNoArms commented:
@stayn могу я слить ветки чтобы делать второе дз?

stayn commented:
да
